common/compositor/gl/glrenderer.cpp

```diff
@@ -21,16 +21,25 @@
 #include "nativesurface.h"
 #include "renderstate.h"
 #include "shim.h"
+#include "hwclayerbuffermanager.h"
 
 namespace hwcomposer {
 
 GLRenderer::~GLRenderer() {
+  if (!context_.MakeCurrent()) {
+    ETRACE("Failed make current context.");
+    return;
+  }
+
+  if (buffer_manager_)
+    buffer_manager_->PurgeGraphicsResources();
```

Comment:
Hmm we do this only when renderer is destroyed, we need to do this every time a buffer is destroyed